portafly/src/components/pages/accounts/listing/AccountsListingTable.tsx

```diff
@@ -0,0 +1,164 @@
+    { name: 'admin', humanName: t('accounts_table.admin_header') },
+    { name: 'state', humanName: t('accounts_table.state_header'), options }
+  ]
+
+  // FIXME: this throws an error
+  // useEffect(resetPagination, [filters])
+
+  const actions = {
+    sendEmail: t('shared:bulk_actions.send_email'),
+    changeState: t('shared:bulk_actions.change_state')
+  }
+
+  // FIXME: this throws an error
+  // const filteredRows = useMemo(() => (
+  //   filterRows(rows, filters, categories, columns)
+  // ), [rows, filters, sortBy])
+
+  const filteredRows = filterRows(rows, filters, columns)
+
+  const visibleRows = filteredRows.slice(startIdx, endIdx)
+
+  const onSort: OnSort = (_event, index, direction) => {
+    setSortBy(index, direction, true)
+  }
+
+  const onSelectOne: OnSelect = (_ev, selected, _rowIndex, rowData) => {
+    selectOne(rowData.id, selected)
+  }
+
+  const pagination = <PaginationWidget itemCount={filteredRows.length} />
+
+  const searchEmptyState = [{
+    heightAuto: true,
+    cells: [{
+      props: { colSpan: 8 },
+      title: <TableEmptyState
+        // title={t('accounts_table.filter_empty_state_title')}
+        title={t('accounts_table.empty_state')}
+        // body={t('accounts_table.filter_empty_state_body')}
+        body={t('accounts_table.empty_state')}
+        button={(
+          <Button variant="link" onClick={() => setFilters({})}>
+            {/* {t('accounts_table.empty_state_button')} */}
+            Clear All Filters
+          </Button>
+        )}
+      />
+    }]
+  }]
+
+  const Header = (
+    <Toolbar>
+      <ToolbarContent>
+        <ToolbarItem>
+          <BulkSelectorWidget />
+        </ToolbarItem>
+        <ToolbarItem>
+          <BulkActionsWidget actions={actions} />
+        </ToolbarItem>
+        <ToolbarItem>
+          <SearchWidget categories={categories} />
+        </ToolbarItem>
+        <ToolbarItem variant="pagination" breakpointMods={[{ modifier: 'align-right', breakpoint: 'md' }]}>
+          {pagination}
+        </ToolbarItem>
+      </ToolbarContent>
+    </Toolbar>
+  )
+
+  return (
+    <>
+      <Table
+        aria-label={t('accounts_table.aria_label')}
+        header={Header}
+        cells={columns}
+        rows={visibleRows.length ? visibleRows : searchEmptyState}
+        onSelect={visibleRows.length ? onSelectOne : undefined}
+        canSelectAll={false}
+        sortBy={sortBy}
+        onSort={onSort}
+      >
+        <TableHeader />
+        <TableBody />
+      </Table>
+      <Toolbar>
+        <ToolbarContent>
+          <ToolbarItem variant="pagination" breakpointMods={[{ modifier: 'align-right', breakpoint: 'md' }]}>
+            {pagination}
+          </ToolbarItem>
+        </ToolbarContent>
+      </Toolbar>
+
+      {modal === 'sendEmail' && (
+        <SendEmailModal
+          items={selectedRows.map((r) => `${(r.cells as string[])[1]} (${(r.cells as string[])[0]})`)}
+        />
+      )}
+
+      {modal === 'changeState' && (
```

Comment:
Maybe this `sendEmail`  and `changeState` can be moved to a CONSTANT

Reply:
`modal` is already typed as `BulkAction` and this type is inferred to `changeState` and `sendEmail`. If we compared anything different, it would throw an error.
What I mean is thanks to Typescript there's no need for Enum or constants.
<img width="752" alt="Screen Shot 2020-06-09 at 11 12 55" src="https://user-images.githubusercontent.com/11672286/84129315-1da2e680-aa42-11ea-8730-421b6e75ba43.png">